File: src/lib/tauri/system.ts
import { invoke } from "@tauri-apps/api/core";
import { TauriBridge } from "@/models";

export type CpuInfo = TauriBridge.System.CpuInfo;
export type MemoryInfo = TauriBridge.System.MemoryInfo;
export type SystemInfo = TauriBridge.System.SystemInfo;

export async function getCpuInfo(): Promise<CpuInfo> {
    return invoke<CpuInfo>("get_cpu_info");
}

export async function getMemoryInfo(): Promise<MemoryInfo> {
    return invoke<MemoryInfo>("get_memory_info");
}

export async function getSystemInfo(): Promise<SystemInfo> {
    return invoke<SystemInfo>("get_system_info");
}

File: src-tauri/src/command/system.rs
use crate::core::system;

#[tauri::command]
pub fn get_cpu_info() -> system::CpuInfo {
    log::info!("[command] get_cpu_info");
    let info = system::cpu_info();
    log::debug!("[command] get_cpu_info: {} cores @ {} MHz", info.core_count, info.frequency);
    info
}

#[tauri::command]
pub fn get_memory_info() -> system::MemoryInfo {
    log::info!("[command] get_memory_info");
    let info = system::memory_info();
    log::debug!("[command] get_memory_info: total={} MB, used={} MB", info.total, info.used);
    info
}

#[tauri::command]
pub fn get_system_info() -> system::SystemInfo {
    log::info!("[command] get_system_info");
    let info = system::system_info();
    log::debug!("[command] get_system_info: os={} {}", info.os_name, info.os_version);
    info
}
